### ISSUE TITLE
The transferred keras file has extra padding layer

### ISSUE THREAD

**Issue by pawopawo** [nerox8664/onnx2keras#38]
![image](https://user-images.githubusercontent.com/38682321/71569663-a3be1700-2b0b-11ea-82e5-31da2fdcca4f.png)

The original onnx file didn't have padding layer, but the extra padding layer was transferred out. 
And onnx file is transferred from pytorch.

![image](https://user-images.githubusercontent.com/38682321/71569669-b0426f80-2b0b-11ea-9bb4-3d6d8a3e44d9.png)

Hope for help~

**Comment by pawopawo** [comment]
```
Deprecated in favor of operator or tf.math.divide.
Unable to use `same` padding. Add ZeroPadding2D layer to fix shapes.
WARNING:tensorflow:From /usr/local/lib/python3.6/site-packages/keras/backend/tensorflow_backend.py:3980: The name tf.nn.avg_pool is deprecated. Please use tf.nn.avg_pool2d instead.

Failed to use keras.layers.Add. Fallback to TF lambda.
Unable to use `same` padding. Add ZeroPadding2D layer to fix shapes.
Failed to use keras.layers.Add. Fallback to TF lambda.
Unable to use `same` padding. Add ZeroPadding2D layer to fix shapes.
Failed to use keras.layers.Add. Fallback to TF lambda.
Failed to use keras.layers.Add. Fallback to TF lambda.
Failed to use keras.layers.Add. Fallback to TF lambda.
Failed to use keras.layers.Add. Fallback to TF lambda.
Failed to use keras.layers.Add. Fallback to TF lambda.
Failed to use keras.layers.Add. Fallback to TF lambda.
Failed to use keras.layers.Add. Fallback to TF lambda.
Failed to use keras.layers.Add. Fallback to TF lambda.
Failed to use keras.layers.Add. Fallback to TF lambda.
Failed to use keras.layers.Add. Fallback to TF lambda.
Unable to use `same` padding. Add ZeroPadding2D layer to fix shapes.
Failed to use keras.layers.Add. Fallback to TF lambda.
Failed to use keras.layers.Add. Fallback to TF lambda.
Failed to use keras.layers.Add. Fallback to TF lambda.
Unable to use `same` padding. Add ZeroPadding2D layer to fix shapes.
Failed to use keras.layers.Add. Fallback to TF lambda.
Failed to use keras.layers.Add. Fallback to TF lambda.
Failed to use keras.layers.Add. Fallback to TF lambda.
Unable to use `same` padding. Add ZeroPadding2D layer to fix shapes.
Failed to use keras.layers.Add. Fallback to TF lambda.
Failed to use keras.layers.Add. Fallback to TF lambda.
Failed to use keras.layers.Add. Fallback to TF lambda.
Unable to use `same` padding. Add ZeroPadding2D layer to fix shapes.
Failed to use keras.layers.Add. Fallback to TF lambda.
Failed to use keras.layers.Add. Fallback to TF lambda.
Failed to use keras.layers.Add. Fallback to TF lambda.
Unable to use `same` padding. Add ZeroPadding2D layer to fix shapes.
Failed to use keras.layers.Add. Fallback to TF lambda.
Failed to use keras.layers.Add. Fallback to TF lambda.
Failed to use keras.layers.Add. Fallback to TF lambda.
Unable to use `same` padding. Add ZeroPadding2D layer to fix shapes.
Failed to use keras.layers.Add. Fallback to TF lambda.
Failed to use keras.layers.Add. Fallback to TF lambda.
WARNING:tensorflow:From /usr/local/lib/python3.6/site-packages/keras/backend/tensorflow_backend.py:1834: The name tf.nn.fused_batch_norm is deprecated. Please use tf.compat.v1.nn.fused_batch_norm instead.

WARNING:tensorflow:From /mnt/cephfs_new_wj/cv/xiaxin/NAS/HR_NAS/pytorch-to-tflite-example/converters.py:37: simple_save (from tensorflow.python.saved_model.simple_save) is deprecated and will be removed in a future version.
Instructions for updating:
This function will only be available through the v1 compatibility library as tf.compat.v1.saved_model.simple_save.
WARNING:tensorflow:From /usr/local/lib/python3.6/site-packages/tensorflow/python/saved_model/signature_def_utils_impl.py:201: build_tensor_info (from tensorflow.python.saved_model.utils_impl) is deprecated and will be removed in a future version.
Instructions for updating:
This function will only be available through the v1 compatibility library as tf.compat.v1.saved_model.utils.build_tensor_info or tf.compat.v1.saved_model.build_tensor_info.
INFO:tensorflow:Assets added to graph.
INFO:tensorflow:No assets to write.
INFO:tensorflow:SavedModel written to: /mnt/cephfs_new_wj/cv/xiaxin/NAS/HR_NAS/pytorch-to-tflite-example/data/saved_model/1/saved_model.pb
Convert saved_model.pb to TFLite model:
WARNING:tensorflow:From /usr/local/lib/python3.6/site-packages/tensorflow/lite/python/convert_saved_model.py:60: load (from tensorflow.python.saved_model.loader_impl) is deprecated and will be removed in a future version.
Instructions for updating:
This function will only be available through the v1 compatibility library as tf.compat.v1.saved_model.loader.load or tf.compat.v1.saved_model.load. There will be a new function for importing SavedModels in Tensorflow 2.0.
WARNING:tensorflow:From /usr/local/lib/python3.6/site-packages/tensorflow/python/training/saver.py:1276: checkpoint_exists (from tensorflow.python.training.checkpoint_management) is deprecated and will be removed in a future version.
Instructions for updating:
Use standard file APIs to check for files with this prefix.
INFO:tensorflow:Restoring parameters from /mnt/cephfs_new_wj/cv/xiaxin/NAS/HR_NAS/pytorch-to-tflite-example/data/saved_model/1/variables/variables
INFO:tensorflow:The given SavedModel MetaGraphDef contains SignatureDefs with the following keys: {'serving_default'}
INFO:tensorflow:input tensors info: 
INFO:tensorflow:Tensor's key in saved_model's tensor_map: input
INFO:tensorflow: tensor name: input_1:0, shape: (-1, 224, 224, 3), type: DT_FLOAT
INFO:tensorflow:output tensors info: 
INFO:tensorflow:Tensor's key in saved_model's tensor_map: category
INFO:tensorflow: tensor name: category_1/BiasAdd:0, shape: (-1, 1000), type: DT_FLOAT
INFO:tensorflow:Restoring parameters from /mnt/cephfs_new_wj/cv/xiaxin/NAS/HR_NAS/pytorch-to-tflite-example/data/saved_model/1/variables/variables
2019-12-30 14:36:50.950925: I tensorflow/core/grappler/devices.cc:60] Number of eligible GPUs (core count >= 8, compute capability >= 0.0): 0 (Note: TensorFlow was not compiled with CUDA support)
2019-12-30 14:36:50.951153: I tensorflow/core/grappler/clusters/single_machine.cc:359] Starting new session
2019-12-30 14:36:51.863254: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:716] Optimization results for grappler item: graph_to_optimize
2019-12-30 14:36:51.863485: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:718]   function_optimizer: function_optimizer did nothing. time = 0.004ms.
2019-12-30 14:36:51.863506: I tensorflow/core/grappler/optimizers/meta_optimizer.cc:718]   function_optimizer: function_optimizer did nothing. time = 0ms.
WARNING:tensorflow:From /usr/local/lib/python3.6/site-packages/tensorflow/lite/python/util.py:238: convert_variables_to_constants (from tensorflow.python.framework.graph_util_impl) is deprecated and will be removed in a future version.
Instructions for updating:
Use `tf.compat.v1.graph_util.convert_variables_to_constants`
WARNING:tensorflow:From /usr/local/lib/python3.6/site-packages/tensorflow/python/framework/graph_util_impl.py:270: extract_sub_graph (from tensorflow.python.framework.graph_util_impl) is deprecated and will be removed in a future version.
Instructions for updating:
Use `tf.compat.v1.graph_util.extract_sub_graph`
INFO:tensorflow:Froze 266 variables.
INFO:tensorflow:Converted 266 variables to const ops.
2019-12-30 14:36:53.832513: I tensorflow/core/grappler/devices.cc:60] Number of eligible GPUs (core count >= 8, compute capability >= 0.0): 0 (Note: TensorFlow was not compiled with CUDA support)
2019-12-30 14:36:53.832828: I tensorflow/core/grappler/clusters/single_machine.cc:359] Starting new session
2019-12-30 14:36:53.946593: E tensorflow/core/grappler/grappler_item_builder.cc:637] Init node 313/kernel/Assign doesn't exist in graph
```


The warning shown above will appear during the convert process

**Comment by pawopawo** [comment]
hope for help

**Comment by nerox8664** [comment]
Hello @pawopawo. Can you please provide onnx model / pytorch model file so I can test and reproduce?

By the way, extra-paddings are used as a work-around for some layers (e.g. ConvTranspose2d and so on) and it's not a big problem if they have zero-width, because it should not affect network result.

**Comment by pawopawo** [comment]
Thank you for your reply!
 The onnx file I used is https://drive.google.com/file/d/1zeF98_RoJ1-L66QBdBhiOYqFvGUvmCmm/view?usp=sharing
